### PR TITLE
date-whitespace-bug-fix

### PR DESCRIPTION
**Pull request by ehanson8**
#### Helpful background context
Though `Marc` is the transform most likely to encounter this issue, I updated the date processing for all of the transforms. Let me know if this overkill.

#### How can a reviewer manually see the effects of these changes?
I uploaded a file with a whitespace date to dev1. Run the following command on `main` to see that the whitespace is in the output and then with `date-whitespace-bug-fix` to see that the date appears without whitespace:

```
pipenv run transform -i s3://timdex-extract-dev-222053980223/alma/date-error.xml -o output/date-error-transformed-records.json -s alma
```

#### Developer
- [x] All new ENV is documented in README
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

#### Includes new or updated dependencies?
NO

### CODE REVIEW

File: tests/test_helpers.py [hunk 1]
```diff
@@ -264,16 +264,21 @@ def test_parse_date_from_string_invalid_date_returns_none():
...
     assert validate_date("circa 1930s", "1234") is False
     assert (
-        "Record # '1234' has a date that couldn't be parsed: circa 1930s"
+        "Record # '1234' has a date that couldn't be parsed: 'circa 1930s'"
     ) in caplog.text
 
 
+def test_validate_date_whitespace_date_logs_error(caplog):
+    assert validate_date("30  ", "1234") is False
+    assert ("Record # '1234' has a date that couldn't be parsed: '30  '") in caplog.text
```
Comment (JPrevost) [inline tests/test_helpers.py:279]: Not directly related to this line, but are we sure OpenSearch is treating "30" as a date the same way it might be meant in MARC? Adding the strip() prevents the problem... but are we ending up with data that matches the intent of the cataloger? I'm not saying we aren't, but since there is no test here to clarify these types of edge cases and whether we want "30" to be valid I'm not sure if we are doing the correct thing by just stripping and not confirming the intent remains for these yet.

Comment (JPrevost) [reply tests/test_helpers.py:279]: Do you have a record id we can compare in OpenSearch with the record in Alma to ensure the two systems are treating two digit dates the same way?

**Review by hakbailey — CHANGES_REQUESTED** [review]
Any time you are using a `NavigableString` as a value, you need to convert it to a regular Python string [per the bs4 docs](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#navigablestring). There's a ticket to make that change across the repo, but you should go ahead and update any code you're already touching, including all of these dates.

File: transmogrifier/sources/datacite.py [hunk 2]
```diff
@@ -118,11 +120,11 @@ def get_optional_fields(self, xml: Tag) -> Optional[dict]:
 
         for date in xml.metadata.find_all("date"):
             d = timdex.Date()
-            if date_value := date.string:
+            if date_value := str(date.string):
```
Comment (hakbailey) [inline transmogrifier/sources/datacite.py:123]: `str(None)` results in `"None"`, which means this will never be False.

File: transmogrifier/sources/datacite.py [hunk 3]
```diff
@@ -108,7 +108,9 @@ def get_optional_fields(self, xml: Tag) -> Optional[dict]:
         # dates
         if publication_year := xml.metadata.find("publicationYear", string=True):
             fields["dates"] = [
-                timdex.Date(kind="Publication date", value=publication_year.string)
+                timdex.Date(
+                    kind="Publication date", value=publication_year.string.strip()
```
Comment (hakbailey) [inline transmogrifier/sources/datacite.py:112]: This is also a NavigableString.

File: transmogrifier/sources/dspace_dim.py [hunk 4]
```diff
@@ -88,11 +88,15 @@ def get_optional_fields(self, xml: Tag) -> Optional[dict]:
 
         # dates
         for date in [d for d in xml.find_all("dim:field", element="date") if d.string]:
```
Comment (hakbailey) [inline transmogrifier/sources/dspace_dim.py:90]: While we're in here, let's simplify this.
```
for date in xml.find_all("dim:field", element="date", string=True):
```

**Review by hakbailey — CHANGES_REQUESTED** [review]
See inline comments.

**Review by hakbailey — APPROVED** [review]
👍
